File: backend/src/main/java/dev/agiro/fanel/calendar/api/CalendarEventDto.java
package dev.agiro.fanel.calendar.api;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;
import java.util.UUID;

public record CalendarEventDto(UUID id, UUID householdId, String title, LocalDate date, LocalDate anchorDate,
                               LocalTime time, Integer durationMinutes, UUID addedBy, List<UUID> assigneeIds,
                               RecurrenceFrequency recurrenceFreq, Integer recurrenceInterval,
                               LocalDate recurrenceUntil) {
}

File: android/app/src/main/java/dev/agiro/fanel/android/data/remote/CalendarEventDto.kt
package dev.agiro.fanel.android.data.remote

data class CalendarEventDto(
    val id: String,
    val householdId: String,
    val title: String,
    val date: String,
    val anchorDate: String?,
    val time: String?,
    val durationMinutes: Int?,
    val addedBy: String?,
    val assigneeIds: List<String>?,
    val recurrenceFreq: String?,
    val recurrenceInterval: Int?,
    val recurrenceUntil: String?
)
